external-crates/move/crates/move-package-alt/src/package/root_package.rs

```diff
@@ -19,6 +23,19 @@ use crate::{
     schema::ParsedLockfile,
 };
 
+#[derive(Debug)]
+enum AddressSource<F: MoveFlavor> {
+    /// Addresses are stored in the `Published.toml` file and retrieved from dependencies' files
+    Published(ParsedPublishedFile<F>),
```

Comment:
We maybe don't need to keep the files in here, since we've already extracted the data into `Self::publication`?

Reply:
actually we keep them so we can replay them when we call save; we could conceivably reread the file but this approach is fine. Maybe add a comment since I've had this conversation with myself before